src/Serializer/Filter/PropertyFilter.php

```diff
@@ -66,4 +66,87 @@ public function getDescription(string $resourceClass): array
             ],
         ];
     }
+
+    /**
+     * Generate an array of whitelist properties to match the format that properties
+     * will have in the request.
+     *
+     * Will transform ['foo', 'group.bar.'baz', 'fuz']
+     * into ['foo', 'group' => ['bar' => 'baz'], 'fuz']
+     *
+     * @param array $whitelist the whitelist to format
+     *
+     * @return array an array containing the whitelist ready to match request parameters
+     */
+    private function formatWhitelist(array $whitelist): array
+    {
+        $results = [];
+        foreach ($whitelist as $propertyPath) {
+            $segments = explode('.', $propertyPath);
+            $root = array_shift($segments);
+            if ($segments) {
+                $segments = array_reverse($segments);
+                $child = [$segments[0]];
+                array_shift($segments);
+                foreach ($segments as $segment) {
+                    $child = [$segment => $child];
+                }
+                $results[$root] = array_merge_recursive($results[$root] ?? [], $child);
+            } else {
+                $results[] = $root;
+            }
+        }
+
+        return $results;
+    }
+
+    /**
+     * Computes the intersection of arrays recursively for arrays with a mix of numeric and associatives keys.
+     *
+     * @param array $array1 the array with master values to check
+     * @param array $array2 An array to compare values against
+     *
+     * @return array an array containing all of the values in array1 recursively whose values exist in array2
+     */
+    private function intersectArrayRecursive(array $array1, array $array2): array
```

Comment:
This function and `partitionAssocAndNumericArrayKeys()` may not really fit into this class, if you want me to extract them into some generic utils class, let me know...

Reply:
Huum, what about something easier (should be tested):

```php
private function getProperties(array $properties, array $whitelist = null)
{
    $whitelist = $whitelist ?? $this->whitelist;
    $result = [];

    foreach ($properties as $key => $value) {
		if (is_numeric($key)) {
            if (in_array($value, $whitelist, true)) {
			    $result[] = $value;
            }

            continue;
		}

        if (isset($whitelist[$key]) && is_array($value) && $recursiveResult = $this->getProperties($value, $whitelist[$key])) {
			$recursiveResult[$key] = $test;
		}
	}

	return $result;
}
```